hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java

```diff
@@ -601,75 +600,81 @@ protected Session session(HugeType tableType) {
-                File dataFile = new File(session.dataPath());
-                fileRenamePairs.add(Pair.of(snapshotFile, dataFile));
-            }
-            /*
-             * NOTE: must close rocksdb instance before deleting file directory,
-             * if close after copying the snapshot directory to origin position,
-             * it may produce dirty data.
-             */
-            for (RocksDBSessions sessions : this.sessions()) {
-                sessions.forceCloseRocksDB();
+            Map<String, RocksDBSessions> snapshotPaths = new HashMap<>();
+            for (Map.Entry<String, RocksDBSessions> entry : this.dbs.entrySet()) {
+                RocksDBSessions sessions = entry.getValue();
+                String snapshotPath = sessions.buildSnapshotPath(snapshotPrefix);
+                LOG.debug("The origin data path: {}", entry.getKey());
+                if (!deleteSnapshot) {
+                    snapshotPath = sessions.hardLinkSnapshot(snapshotPath);
+                }
+                LOG.debug("The snapshot data path: {}", snapshotPath);
+                snapshotPaths.put(snapshotPath, sessions);
             }
-            // Copy snapshot file to dest file
-            for (Pair<File, File> pair : fileRenamePairs) {
-                File snapshotFile = pair.getLeft();
-                File dataFile = pair.getRight();
-                try {
-                    if (dataFile.exists()) {
-                        LOG.warn("Delete origin data directory {}", dataFile);
-                        FileUtils.deleteDirectory(dataFile);
+
+            for (Map.Entry<String, RocksDBSessions> entry :
+                 snapshotPaths.entrySet()) {
+                String snapshotPath = entry.getKey();
+                RocksDBSessions sessions = entry.getValue();
+                sessions.resumeSnapshot(snapshotPath);
+
+                if (deleteSnapshot) {
+                    // Delete empty snapshot parent directory
+                    Path parentPath = Paths.get(snapshotPath).getParent();
+                    if (Files.list(parentPath).count() == 0) {
+                        FileUtils.deleteDirectory(parentPath.toFile());
                     }
-                    FileUtils.moveDirectory(snapshotFile, dataFile);
-                } catch (IOException e) {
-                    throw new BackendException("Failed to move %s to %s",
-                                               e, snapshotFile, dataFile);
                 }
             }
-            // Reload rocksdb instance
-            for (RocksDBSessions sessions : this.sessions()) {
-                sessions.reload();
-            }
-            LOG.info("The store {} load snapshot successfully", this.store);
-        } catch (RocksDBException e) {
-            throw new BackendException("Failed to reload rocksdb", e);
+            LOG.info("The store '{}' resume snapshot successfully", this.store);
```

Comment:
log this instead of this.store, let output String.format("%s/%s", this.database(), this.store())